frontend/src/lib/StaticGraphParser.ts

```diff
@@ -263,3 +264,60 @@ export function createGraph(workflow: Workflow): dagre.graphlib.Graph {
 
   return graph;
 }
+
+/**
+ * Perform a transitive reduction over the input graph.
+ *
+ * From [1]: Transitive reduction of a directed graph D is another directed
+ * graph with the same vertices and as few edges as possible, such that for all
+ * pairs of vertices v, w a (directed) path from v to w in D exists if and only
+ * if such a path exists in the reduction
+ *
+ * The current implementation has a time complexity bound `O(n*m)`, where `n`
+ * are the nodes and `m` are the edges of the input graph.
+ *
+ * [1]: https://en.wikipedia.org/wiki/Transitive_reduction
+ *
+ * @param graph The dagre graph object
+ */
+export function transitiveReduction(graph: dagre.graphlib.Graph) {
```

Comment:
Right now, we do not enforce explicit return value type when it can be auto inferred.

Can you elaborate rationale of your preference? We can consider that.

Reply:
Defining a return type allows IDE to perform `type check` and make right suggestion to usage. For example: IDE will suggest to add `?` if the variable can be `undefined`. And the other method `createGraph` in the same file does specify the return type.

I think having a return type specified is a way to leverage the power of typescript (when considering typescript vs javascript). But since it is not the current practice, we don't need to enforce them in this PR, we can discuss in other channel.

Reply:
@zijianjoy IDE no longer requires return type for type checking, the typescript language server has all the information available at compile time, so inferred return type also works.

Reply:
https://www.typescriptlang.org/docs/handbook/type-inference.html
In some rare cases, inferred type may not be what we want -- that's when we "must" specify a return type.

Reply:
but it's anyway a valid preference, if you think explicit return type makes code review easier

Reply:
Thank you for the link to type inference! I understand the situation to specify return type when inference is not what we want now. Sounds good on keeping this practice.